### PR TITLE
Add mongo db connection

### PR DESCRIPTION
**Pull request by longnghiem**

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -1,27 +1,28 @@
...
-		"no-var": "error",
-		"comma-dangle": ["error", "always"],
-		"no-multiple-empty-lines": ["error"],
-		"no-trailing-spaces": ["error"],
-		"no-undef": ["error"],
-		"prefer-template": "error",
-		"no-unused-vars": "error",
-		"no-console": ["warn"]
-	}
+  "plugins": ["prettier"],
```
Comment (ximbong91023) [inline .eslintrc.json]: Add at the bottom

File: app.js [hunk 2]
```diff
@@ -8,19 +8,26 @@ const mongoose = require("mongoose");
 const indexRouter = require("./routes/index");
 const usersRouter = require("./routes/users");
 
-/* const keys = require("./config/keys"); */
+const keys = require("./config/keys");
 
 const app = express();
 
+console.log("we are now using: ", keys.mongoURI);
```
Comment (ximbong91023) [inline app.js]: Delete this line

File: app.js [hunk 3]
```diff
@@ -8,19 +8,26 @@ const mongoose = require("mongoose");
...
 	{
-		dbName: "admin",
 		useNewUrlParser: true,
 	}
 );
 
+const db = mongoose.connection;
+db.on("error", console.error.bind(console, "connection error:"));
+db.once("open", () => {
+	console.log("we are connected");
```
Comment (ximbong91023) [inline app.js]: Add URI here

File: config/keys.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
+const prod_mongoURI = require("./prod");
```
Comment (ximbong91023) [inline config/keys.js]: Rename this variable

File: package.json [hunk 5]
```diff
@@ -21,6 +21,14 @@
     "ejs": "~2.5.7",
     "express": "~4.16.0",
     "http-errors": "~1.6.2",
+    "mongoose": "^5.2.8",
     "morgan": "~1.9.0"
+  },
+  "devDependencies": {
```
Comment (ximbong91023) [inline package.json]: Why do we need this?

Comment (longnghiem) [reply package.json]: we cannot install other plugins and configs alone, so we need to manually install eslint.

File: .eslintrc.json (hunk 1 above)
Comment (longnghiem) [reply .eslintrc.json]: yes, my bad.

Comment (longnghiem) [reply .eslintrc.json]: but actually, I think all the plugins should be stated on the top.

File: app.js [hunk 6]
```diff
@@ -3,12 +3,29 @@ const express = require("express");
...
+const mongoose = require("mongoose");
 
 const indexRouter = require("./routes/index");
 const usersRouter = require("./routes/users");
 
+const keys = require("./config/keys");
+
 const app = express();
 
+//mongoDB connection
```
Comment (Nizar-Rahme) [inline app.js]: Test

File: package.json (hunk 5 above)
Comment (ximbong91023) [reply package.json]: I still don't get the point of using these packages.

**Comment by ximbong91023** [comment]
Looks good

**Pull request merged**